packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -671,3 +671,45 @@ class DrivenScrollActivity extends ScrollActivity {
     return '${describeIdentity(this)}($_controller)';
   }
 }
+
+//// An activity that a scroll view based on delta
+////
+//// For example, a [PointerScrollActivity] is used to implement
+//// [ScrollPositionWithSingleContext.pointerScroll].
+class PointerScrollActivity extends ScrollActivity {
+
+  /// Creates an activity that a scroll view based on delta.
+  ///
+  /// All of the parameters must be non-null.
+  PointerScrollActivity(
+      ScrollActivityDelegate delegate,
+      this.delta,
+      ) : super(delegate) {
+    scheduleMicrotask(() {
+      delegate?.applyUserOffset(delta);
+      _isScrolling = false;
+      delegate?.goBallistic(0);
+    });
+  }
+
+  /// The offset of the scroll.
+  /// It will also act as [velocity].
+  final double delta;
+
+  //
+  bool _isScrolling = true;
+
+  @override
+  void dispatchOverscrollNotification(ScrollMetrics metrics, BuildContext context, double overscroll) {
+    OverscrollNotification(metrics: metrics, context: context, overscroll: overscroll, velocity: delta).dispatch(context);
```

Comment:
This seems wrong. A pointer scroll does not have a velocity, and should not trigger overscroll.

If you're trying to solve the problem that trackpad scroll gestures don't overscroll, this is not the way to do it; the reason for that is that OS-sourced gesture support (trackpad scroll, pinch, etc.) hasn't been implemented yet. The fact that trackpad scrolls currently send pointer events is a workaround for that currently-missing functionality, not the long-term solution.

Reply:
Understood, I'm ignoring the difference between the mouse wheel and the touchpad. I will make the changes.